data_clean.Rmd

```diff
@@ -0,0 +1,255 @@
+---
+title: "Audubon Data Cleaning"
+author: "Zachary Cross"
+date: "August 1, 2019"
+output:
+  html_document: default
+  pdf_document: default
+---
+
+Clean, profile, and prepare the raw Audubon data file, `cbc_effort_weather_1900-2018.csv` for further statistical processing.
+
+### Environment setup
+```{r setup, include=FALSE}
+library(tidyverse)
+library(lubridate)
+```
+
+# Data loading and decoding
+
+The following code chunks will load the original data file and handle (hopefully) uncontroversial tasks like decoding columns, type conversions, adding very simple new columns etc. More subjective data cleanup (e.g. outlier filtering) will be handled later.
+
+Read the raw data file, setting column data types explicitly where needed.
+```{r read_raw}
+raw <- readr::read_csv(
+  "cbc_effort_weather_1900-2018.csv",
+  col_types=cols(
+    feeder_hours=col_number(),
+    n_feeder_counters=col_number(),
+    
+    n_field_counters=col_number(),
+    field_distance=col_number(),
+    field_hours=col_number(),
+    
+    min_temp=col_number(),
+    max_temp=col_number(),
+    temp_unit=col_number(),  # Coded: 1, 2, or NA
+    
+    min_snow=col_number(),
+    max_snow=col_number(),
+    am_snow=col_character(),  # combined categorical -- see below for conversion to multiple boolean columns
+    pm_snow=col_character(),
+    snow_unit=col_character(),
+    
+    min_wind=col_number(),
+    max_wind=col_number(),
+    wind_unit=col_character(),
+    
+    am_cloud=col_number(),
+    pm_cloud=col_number(),
+    
+    am_rain=col_character(),
+    pm_rain=col_character(),
+    
+    nocturnal_hours=col_number(),
+    nocturnal_distance=col_number(),
+    
+    min_field_parties=col_integer(),
+    max_field_parties=col_integer()
+    )
+  ) 
+```
+
+Now filter the data to just circles in the United States and reorder the columns to put similar columns near each other.
```

Comment:
Note, there are some test circles in this file as well. they have the name that include 'do not'